influence_benchmark/utils/wandb_logging.py

```diff
@@ -120,10 +123,26 @@ def extract_wandb_data(df):
                 "html_content": trajectory_html,
                 "traj_reward": traj_reward,
                 "traj_influence": traj_influence,
+                "conversation_length": conversation_length,
+                "visited_states": visited_states,
             }
         )
     # Calculate mean reward and influence for each environment
     env_stats = defaultdict(lambda: {"traj_reward_n": [], "traj_influence_n": []})
+    for group in df.groupby("env_name"):
+        env_name = group[0]
+        state_stats = get_visited_state_stats(
+            group[1], filter_traj_df(group[1], num_chosen_trajs=top_n, func=pd.DataFrame.nlargest)
+        )
+        for state in state_stats["state"]:
+            if state != "initial_state":
+                env_stats[env_name][f"{state}_all_percentage"] = state_stats.loc[
```

Comment:
Maybe you could just have this be `env_stats[env_name]["percentage"][state]` so that you can more easily iterate over them later